src/crawlers/basic_crawler.js

```diff
@@ -485,32 +489,33 @@ class BasicCrawler {
     /**
      * Handles errors thrown by user provided handleRequestFunction()
      * @param {Error} error
-     * @param {Request} request
+     * @param {object} crawlingContext
      * @param {(RequestList|RequestQueue)} source
      * @return {Promise<boolean|void|QueueOperationInfo>} willBeRetried
      * @ignore
      */
-    async _requestFunctionErrorHandler(error, request, source) {
-        request.pushErrorMessage(error);
+    async _requestFunctionErrorHandler(error, crawlingContext, source) {
+        crawlingContext.request.pushErrorMessage(error);
```

Comment:
```js
const { request } = crawlingContext;
request.pushErrorMessage(error);
// ...
```
Would look better imo

Reply:
Sure. Just have to add something like ` crawlingContext.request = request;` before calling the `handleFailedRequestFunction` because of the reference right?